File: algorithms/1614.maximum-nesting-depth-of-the-parentheses/solution.rb
 # @param {String} s
# @return {Integer}
def max_depth(s)
  max, current = 0, 0

  s.each_char do |c|
    if c == "("
      current += 1
      max = [max, current].max
    elsif c == ")"
      current -= 1 if current > 0
    end
  end

  max
end

File: algorithms/1614.maximum-nesting-depth-of-the-parentheses/solution.go
func maxDepth(s string) int {
	max := 0
	current := 0

	for _, c := range s {
		if c == '(' {
			current += 1
			if current > max {
				max = current
			}
		} else if c == ')' {
			if current > 0 {
				current -= 1
			}
		}
	}

	return max
}
